src/apps/chifra/pkg/tslib/tslib.go

```diff
@@ -0,0 +1,143 @@
+package tslibPkg
+
+import (
+	"encoding/binary"
+	"errors"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+const tsPath = "/Users/jrush/Development/trueblocks-index/wild.0/unchained/ts.bin"
+
+type Timestamp struct {
+	Bn uint32 `json:"bn"`
+	Ts uint32 `json:"ts"`
+}
+
+var recordCount uint64 = 0
+
+func nRecords() (uint64, error) {
+	if recordCount > 0 {
+		return recordCount, nil
+	}
+	fileStat, err := os.Stat(tsPath)
+	if err != nil {
+		return 0, err
+	}
+	recordCount = uint64(fileStat.Size()) / 8
+	return recordCount, nil
+}
+
+var loaded = false
```

Comment:
We should discuss this (in these conversations, I suppose).

What do you think about this 'global' memory?

It makes the command-line code very significantly faster if we're converting 100,000s of timestamps in a single invocation of this function (which happens), but it seems a bit precarious in the server code.

I feel like this "issue" will keep coming up (I see it in the `--neighbor` option to `chifra export` as well) and in the names database, so a "pattern" now is better than a pattern later.

Reply:
I don't see any problem with a global cache other that filling it with values. This would be problematic in the server, because each goroutine could try to initialize the cache.

I think that we could use something like `sync.Cond` to make the goroutines wait until the cache is populated with values: https://stackoverflow.com/a/42772799
What do you think?

Reply:
I'll merge this and make an issue so we don't forget.